qcelemental/models/results.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+from pydantic import BaseModel, constr
+from typing import List, Union
+from .molecule import Molecule
+from .common_models import Provenance
+from ..util import provenance_stamp
+
+
+class DriverEnum(str, Enum):
+    energy = 'energy'
+    gradient = 'gradient'
+    hessian = 'hessian'
+
+
+class Model(BaseModel):
+    method: str
+    basis: str = None
+    # basis_spec: BasisSpec = None  # This should be exclusive with basis, but for now will be omitted
+
+
+class Properties(BaseModel):
+    scf_one_electron_energy: float = None
+    scf_two_electron_energy: float = None
+    nuclear_repulsion_energy: float = None
+    scf_vv10_energy: float = None
+    scf_xc_energy: float = None
+    scf_dispersion_correction_energy: float = None
+    scf_dipole_moment: List[float] = None
+    scf_total_energy: float = None
+    scf_iterations: int = None
+    mp2_same_spin_correlation_energy: float = None
+    mp2_opposite_spin_correlation_energy: float = None
+    mp2_singles_energy: float = None
+    mp2_doubles_energy: float = None
+    mp2_total_correlation_energy: float = None
+    mp2_total_energy: float = None
+    calcinfo_nbasis: int = None
+    calcinfo_nmo: int = None
+    calcinfo_nalpha: int = None
+    calcinfo_nbeta: int = None
+    calcinfo_natom: int = None
+    return_energy: float = None
+
+    class Config:
+        allow_extra = False
+
+
+class ErrorEnum(str, Enum):
+    convergence_error = "convergence_error"
+    file_error = "file_error"
+    memory_error = "memory_error"
```

Comment:
Not yet strict enum.

Reply:
The schema makes it a strict enum at the moment, but we could convert it back to a `Dict[str, str]` for now

Reply:
Please do.